persistence/src/main/java/cz/fi/muni/pa165/entity/MushroomCount.java

```diff
@@ -14,7 +14,7 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
-    @ManyToOne(optional = false)
+    @ManyToOne(optional = false, cascade = CascadeType.PERSIST)
```

Comment:
I don't think this is a good idea. MushroomCount shouldn't care about persisting mushroom